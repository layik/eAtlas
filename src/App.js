/**
 * 
 */
import React, { Component } from 'react';
import { Provider as StyletronProvider } from 'styletron-react';
import { BaseProvider, DarkTheme } from 'baseui';
import { Client as Styletron } from 'styletron-engine-atomic';

import Welcome from './Welcome';

import './App.css';

const engine = new Styletron();

class App extends Component {
  render() {    
    return (
      <main>
        <StyletronProvider value={engine}>
          <BaseProvider theme={DarkTheme}>
            <Welcome 
            //TODO or get it from a settings-json file/objeect
            defaultURL={this.props.defaultURL} 
            data={this.props.data}/>
          </BaseProvider>
        </StyletronProvider>
      </main>
    )
  }
}

export default App;
