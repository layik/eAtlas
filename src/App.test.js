import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'; 
import { shallow } from 'enzyme';
import { BaseProvider, DarkTheme } from 'baseui';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('App - should create snapshot', () => { 
  const component = renderer.create( 
    <BrowserRouter><App /></BrowserRouter> 
  ); 
  const tree = component.toJSON(); 
  expect(tree).toMatchSnapshot(); 

  const m = shallow(<App />);
  console.log(m.find(BaseProvider).children());
  expect(m.find(BaseProvider).prop('theme')).toEqual(DarkTheme);
})