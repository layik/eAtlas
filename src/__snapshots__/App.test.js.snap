// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - should create snapshot 1`] = `
<main>
  <div>
    <div>
      <div
        className="loader"
        style={
          Object {
            "visibility": "visible",
            "zIndex": -1,
          }
        }
      />
      <div
        style={
          Object {
            "cursor": "grab",
            "height": "768px",
            "position": "relative",
            "width": "1024px",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            className=""
            style={
              Object {
                "height": "100%",
                "overflow": "hidden",
                "position": "absolute",
                "visibility": "inherit",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "height": 0,
                "overflow": "visible",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        className="side-panel-container"
        style={
          Object {
            "marginLeft": 0,
          }
        }
      >
        <div
          className="side-panel"
        >
          <div
            className="side-pane-header"
          >
            <h2>
              Nothing to show
            </h2>
            dataset: 
          </div>
          <div>
            <button
              className="ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az b0 b1 b2 b3 b4 b5 b6 b7 b8 b9 ba bb bc bd"
              data-baseweb="button"
              disabled={false}
              onClick={[Function]}
            >
              Add data
            </button>
            <i
              className="fa fa-table"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                  "margin": 5,
                }
              }
            />
          </div>
          <div
            className="side-panel-body"
          >
            <div
              className="side-panel-body-content"
            >
              <br />
              <hr
                style={
                  Object {
                    "clear": "both",
                  }
                }
              />
              <div
                id="main-tabs"
              >
                <ul
                  className="nav nav-tabs"
                  role="tablist"
                >
                  <li
                    className="active"
                    role="presentation"
                  >
                    <a
                      aria-controls="main-tabs-pane-1"
                      aria-selected={true}
                      href="#"
                      id="main-tabs-tab-1"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="tab"
                    >
                      <i
                        className="fa fa-info"
                        style={
                          Object {
                            "fontSize": "2rem",
                          }
                        }
                      />
                    </a>
                  </li>
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      aria-controls="main-tabs-pane-2"
                      aria-selected={false}
                      href="#"
                      id="main-tabs-tab-2"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="tab"
                      tabIndex={-1}
                    >
                      <i
                        className="fa fa-sliders"
                        style={
                          Object {
                            "fontSize": "2rem",
                          }
                        }
                      />
                    </a>
                  </li>
                  <li
                    className=""
                    role="presentation"
                  >
                    <a
                      aria-controls="main-tabs-pane-3"
                      aria-selected={false}
                      href="#"
                      id="main-tabs-tab-3"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="tab"
                      tabIndex={-1}
                    >
                      <i
                        className="fa fa-filter"
                        style={
                          Object {
                            "fontSize": "2rem",
                          }
                        }
                      />
                    </a>
                  </li>
                </ul>
                <div
                  className="tab-content"
                >
                  <div
                    aria-hidden={false}
                    aria-labelledby="main-tabs-tab-1"
                    className="fade tab-pane active in"
                    id="main-tabs-pane-1"
                    role="tabpanel"
                  />
                  <div
                    aria-hidden={true}
                    aria-labelledby="main-tabs-tab-2"
                    className="fade tab-pane"
                    id="main-tabs-pane-2"
                    role="tabpanel"
                  >
                    Map Styles
                    <br />
                    <div
                      className="dropdown btn-group"
                    >
                      <button
                        aria-expanded={false}
                        aria-haspopup={true}
                        className="dropdown-toggle btn btn-default"
                        disabled={false}
                        id="dropdown-size-medium"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                        type="button"
                      >
                        Default(dark)
                         
                        <span
                          className="caret"
                        />
                      </button>
                      <ul
                        aria-labelledby="dropdown-size-medium"
                        className="dropdown-menu"
                        role="menu"
                      >
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Dark
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Basic
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Streets
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Bright
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Light
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            Satellite
                          </a>
                        </li>
                        <li
                          className="mi-class"
                          role="presentation"
                        >
                          <a
                            href="#"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            No map
                          </a>
                        </li>
                      </ul>
                    </div>
                    <div
                      className="checkbox"
                    >
                      <label
                        title=""
                      >
                        <input
                          disabled={false}
                          onChange={[Function]}
                          type="checkbox"
                        />
                        Hex Plot
                      </label>
                    </div>
                    <div
                      className="checkbox"
                    >
                      <label
                        title=""
                      >
                        <input
                          disabled={false}
                          onChange={[Function]}
                          type="checkbox"
                        />
                        Subset by map boundary
                      </label>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    aria-labelledby="main-tabs-tab-3"
                    className="fade tab-pane"
                    id="main-tabs-pane-3"
                    role="tabpanel"
                  />
                </div>
              </div>
            </div>
            <div
              className="space"
            />
            <form
              className="search-form"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <span
                  className="input-group"
                >
                  <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="fly to..."
                    type="text"
                  />
                  <span
                    className="input-group-addon"
                  >
                    <span
                      className="glyphicon glyphicon-search"
                    />
                  </span>
                </span>
              </div>
            </form>
          </div>
        </div>
        <div
          className="close-button"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "#242730",
              }
            }
          >
            <i
              className="fa fa-arrow-circle-left"
              style={
                Object {
                  "color": "white !important",
                  "fontSize": "2rem",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</main>
`;
